### PR TITLE
Docs: Add Find Command Implementation Details

### LINKED ISSUE #113: Docs: Add UML diagram for find command
Referencing #110.

### PR DESCRIPTION
**Pull request by RyanNgWH**
Added implementation details & steps to define/redefine how a field is being searched to the developer guide

Closes #113

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -214,6 +214,20 @@ The diagram highlights the four main components of FitBook, highlighted in their
 
 > The above sequence diagram also applies to the removal of a note from an existing client when no input string or prefix is entered for the `note` command. (i.e. `note 1`, or `note 1 note/`).
 
+### Searching Clients
+Search for clients are done using the `find` command. The command has been designed to be extendable, allowing for developers to easily define how new fields (attributes) in the clients can be searched.
+
+The sequence diagram below shows the logic flow of executing the command `find n/wendy`
```
Comment (ruijietay) [inline docs/DeveloperGuide.md:220]:
```suggestion
The sequence diagram below shows the logic flow of executing the command `find n/wendy`.
```
Just one tiny nit!

**Review by ruijietay — APPROVED** [review]
Just one tiny nit! Apart from that, great work! 😄

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (RyanNgWH) [reply docs/DeveloperGuide.md]: Oh nice catch! 😄

**Review by dillontkh — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -214,6 +214,20 @@ The diagram highlights the four main components of FitBook, highlighted in their
 
 > The above sequence diagram also applies to the removal of a note from an existing client when no input string or prefix is entered for the `note` command. (i.e. `note 1`, or `note 1 note/`).
 
+### Searching Clients
+Search for clients are done using the `find` command. The command has been designed to be extendable, allowing for developers to easily define how new fields (attributes) in the clients can be searched.
```
Comment (bgopi23) [inline docs/DeveloperGuide.md:218]:
```suggestion
Search for clients is done using the `find` command. The command has been designed to be extendable, allowing for developers to easily define how new fields (attributes) in the clients can be searched.
```

Comment (bgopi23) [reply docs/DeveloperGuide.md:218]: just a minor nit

Comment (RyanNgWH) [reply docs/DeveloperGuide.md]: Good catch! Thanks!

**Pull request merged**